src/deepsparse/cpu.py

```diff
@@ -140,6 +143,51 @@ def _parse_arch_bin() -> architecture:
         raise OSError(error_msg.format(ex))
 
 
+def allow_darwin() -> bool:
+    """
+    Check if Darwin support is allowed.
+    """
+    return int(os.getenv("NM_ALLOW_DARWIN", 0)) != 0
+
+
+def get_darwin_version() -> str:
+    """
+    If we are running Darwin, get the current version.  Otherwise return None.
+    """
+    if sys.platform.startswith("darwin"):
+        return platform.mac_ver()[0]
+    return None
+
+
+def check_darwin_support() -> bool:
+    """
+    Check if the system is running Darwin and it meets the minimum version
+    requirements.
+    """
+    if sys.platform.startswith("darwin"):
```

Comment:
Should we be failing the darwin support check if `NM_ALLOW_DARWIN` is `False`?
```suggestion
    if allow_darwin() and sys.platform.startswith("darwin"):
```

Reply:
Good catch, I missed this while doing refactoring.